# COCO Train set


In [1]:
from pycocotools.coco import COCO
import os

In [2]:
dsNm = 'train2017'
root = 'E:\Resource\Dataset\COCO'
annDir = os.path.join(root,'annotations','instances_{}.json'.format(dsNm))

# Initialize coco api
coco = COCO(annDir)

loading annotations into memory...
Done (t=26.59s)
creating index...
index created!


## Record of Images Number for different Cats

I select bicycle, car, motorcycle, bus, truck

In [3]:
# load cats
catNms = ['bicycle', 'car', 'motorcycle', 'bus', 'truck']
cats = coco.loadCats(coco.getCatIds(catNms=catNms))
count = []
for cat in cats:
    imgIds = coco.getImgIds(catIds=cat['id'])
    count = count + imgIds
    count = list(dict.fromkeys(count))
    print(cat['name'],': ',len(imgIds))
print('Total number of images:',len(count))

bicycle :  3252
car :  12251
motorcycle :  3502
bus :  3952
truck :  6127
Total number of images: 19759


|    Cat's Nm | Amount |
| ----------: | -----: |
|      person |  64115 |
|         car |  12251 |
| person and car |   8519 |
| persion or car | 67847 |

In [8]:
# select categories
catNms = ['person','car']
# get catIds
catIds = coco.getCatIds(catNms=catNms)
print("Categories IDs:",catIds)
# get ImgIds
imgIds = coco.getImgIds(catIds=catIds)
# count imgs number
print('Intersection:',len(imgIds))

imgIds = []
for catId in catIds:
    imgIds = imgIds + coco.getImgIds(catIds=[catId])
imgIds = list(dict.fromkeys(imgIds))  # remove duplicate id
print('Union:',len(imgIds))

Categories IDs: [1, 3]
Intersection: 8519
Union: 67847


In [13]:
cats = coco.loadCats(coco.getCatIds())
for cat in cats:
    idx = cat['id']
    name = cat['name']
    count = len(coco.getImgIds(catIds=idx))
    print('{}.{}:{}'.format(idx, name, count))

1.person:64115
2.bicycle:3252
3.car:12251
4.motorcycle:3502
5.airplane:2986
6.bus:3952
7.train:3588
8.truck:6127
9.boat:3025
10.traffic light:4139
11.fire hydrant:1711
13.stop sign:1734
14.parking meter:705
15.bench:5570
16.bird:3237
17.cat:4114
18.dog:4385
19.horse:2941
20.sheep:1529
21.cow:1968
22.elephant:2143
23.bear:960
24.zebra:1916
25.giraffe:2546
27.backpack:5528
28.umbrella:3968
31.handbag:6841
32.tie:3810
33.suitcase:2402
34.frisbee:2184
35.skis:3082
36.snowboard:1654
37.sports ball:4262
38.kite:2261
39.baseball bat:2506
40.baseball glove:2629
41.skateboard:3476
42.surfboard:3486
43.tennis racket:3394
44.bottle:8501
46.wine glass:2533
47.cup:9189
48.fork:3555
49.knife:4326
50.spoon:3529
51.bowl:7111
52.banana:2243
53.apple:1586
54.sandwich:2365
55.orange:1699
56.broccoli:1939
57.carrot:1683
58.hot dog:1222
59.pizza:3166
60.donut:1523
61.cake:2925
62.chair:12774
63.couch:4423
64.potted plant:4452
65.bed:3682
67.dining table:11837
70.toilet:3353
72.tv:4561
73.laptop:3524
74.mou

In [ ]:
# only use one class, just as Penn-Fudan Demo did. For example, dog.
